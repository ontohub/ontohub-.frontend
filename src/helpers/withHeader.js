import React from 'react'

export const withHeader = (Header, Component) => (props) => (
  <div>
    <Header {...props} />
    <Component {...props} />
  </div>
)

export default withHeader
