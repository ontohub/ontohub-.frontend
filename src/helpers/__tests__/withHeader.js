import React from 'react'
import { withHeader } from '../withHeader'
import { shallow } from 'enzyme'

describe('withHeader', () => {
  const Component = withHeader('h1', 'p'),
        wrapper = shallow(<Component />)
  it('matches the snapshot', () => {
    expect(wrapper.getNode()).toMatchSnapshot()
  })

  it('renders the header and the content', () => {
    expect(wrapper.find('h1').length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })
})
