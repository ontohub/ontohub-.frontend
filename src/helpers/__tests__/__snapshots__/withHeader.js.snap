// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withHeader matches the snapshot 1`] = `
<div>
  <h1 />
  <p />
</div>
`;
